rclcpp/test/test_function_traits.cpp

```diff
@@ -142,15 +143,15 @@ TEST(TestFunctionTraits, arity) {
       return 0;
     };
 
-  auto lambda_one_int = [](int) {
+  auto lambda_one_int = [](int /*one*/) {
```

Comment:
Two problems with this particular line. One is that cpplint think that is a C-Style cast, which it isn't, but the other issue is that cpplint asserts that you should put the parameters into function prototypes. Examples:

```
/Users/william/ros2/src/ros2/rclcpp/rclcpp/test/test_function_traits.cpp:24:  Using C-style cast.  Use static_cast<int>(...) instead  [readability/casting] [4]
```

```
/Users/william/ros2/src/ros2/rclcpp/rclcpp/test/test_function_traits.cpp:431:  All parameters should be named in a function  [readability/function] [3]
```

So unless I suppress those warnings, it isn't possible to leave out the parameter name.